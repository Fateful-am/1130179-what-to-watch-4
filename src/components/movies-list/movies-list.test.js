import React from 'react';
import renderer from 'react-test-renderer';
import MoviesList from './movies-list';

it(`Render Movies List`, () => {
  const tree = renderer
    .create(
        <MoviesList
          movies={[{
            id: 4,
            title: `Moonrise Kingdom`,
            previewSource: `img/moonrise-kingdom.jpg`,
            previewMovie: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
          },
          {
            id: 5,
            title: `Seven Years in Tibet`,
            previewSource: `img/seven-years-in-tibet.jpg`,
            previewMovie: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
          },
          {
            id: 6,
            title: `Midnight Special`,
            previewSource: `img/midnight-special.jpg`,
            previewMovie: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
          },
          {
            id: 7,
            title: `War of the Worlds`,
            previewSource: `img/war-of-the-worlds.jpg`,
            previewMovie: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
          },
          ]}
          onMovieCardClick={()=>{}}
        />, {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
