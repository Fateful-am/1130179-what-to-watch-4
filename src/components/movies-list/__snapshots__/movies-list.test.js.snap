// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Movies List 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    id="mc-4"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster="img/moonrise-kingdom.jpg"
        preload="none"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        Moonrise Kingdom
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id="mc-5"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster="img/seven-years-in-tibet.jpg"
        preload="none"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        Seven Years in Tibet
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id="mc-6"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster="img/midnight-special.jpg"
        preload="none"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        Midnight Special
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    id="mc-7"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster="img/war-of-the-worlds.jpg"
        preload="none"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        War of the Worlds
      </a>
    </h3>
  </article>
</div>
`;
